src/main/java/com/zendesk/maxwell/producer/MaxwellNatsProducer.java

```diff
@@ -0,0 +1,88 @@
+package com.zendesk.maxwell.producer;
+
+import com.zendesk.maxwell.MaxwellContext;
+import com.zendesk.maxwell.replication.Position;
+import com.zendesk.maxwell.row.RowMap;
+import io.nats.streaming.AckHandler;
+import io.nats.streaming.StreamingConnection;
+import io.nats.streaming.StreamingConnectionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Nats Streaming producer
+ */
+public class MaxwellNatsProducer extends AbstractAsyncProducer {
+	private static final Logger logger = LoggerFactory.getLogger(MaxwellNatsProducer.class);
+	private final String natsSubject;
+	private StreamingConnection connection;
+
+	public MaxwellNatsProducer(MaxwellContext context, String natsSubject) {
+		super(context);
+		this.natsSubject = natsSubject;
+		StreamingConnectionFactory cf = new StreamingConnectionFactory(context.getConfig().natsClusterId, context.getConfig().natsClientId);
+		cf.setNatsUrl(context.getConfig().natsUrl);
+		try {
+			this.connection = cf.createConnection();
+		} catch (IOException | InterruptedException e) {
+			logger.error("Exception during Nats Connection", e);
+		}
+	}
+
+
+	@Override
+	public void sendAsync(RowMap r, CallbackCompleter cc) throws Exception {
+		String msg = r.toJSON();
+		NatsAckHandler handler = new NatsAckHandler(cc, r.getNextPosition(), msg, context);
```

Comment:
you probably mean `getPosition` not `getNextPosition`

Reply:
When I check the usage of getNextPosition in Producers such as MaxwellKinesisProducer, MaxwellKafkaProducer (classes extending AbstractAsyncProducer) all of them use getNextPosition. Redis producer is also using the same.

Reply:
yes, they use `getNextPosition` to advance the binlog, but if you're just logging the position of the event, that position is `getPosition`....